src/main/java/com/tagtraum/perf/gcviewer/view/ModelChartImpl.java

```diff
@@ -185,6 +184,24 @@ private void maybePopup(MouseEvent e) {
             }
         });
 
+
+        addMouseWheelListener(new MouseWheelListener() {
+            @Override
+            public void mouseWheelMoved(MouseWheelEvent e) {
+                if(GCViewerGui.isShiftPressed()) {
+                    double pos = (double)(getHorizontalScrollBar().getValue())/chart.getWidth();
+                    if(e.getWheelRotation() > 0 && getScaleFactor() < 100) {
+                        setScaleFactor((getScaleFactor()*1.2));
+                    }
+                    if(e.getWheelRotation() < 0 && getScaleFactor() > 0.01) {
+                        setScaleFactor((getScaleFactor()/1.2));
+                    }
```

Comment:
There's a possibility that `getWheelRotation()` would return `0`.
See https://github.com/apache/jmeter/commit/38eb2be8fb147461f47709f2c85e957518099edb#diff-c29d87213ad842384446cdab147eb08eec0e24362862aec545e541a957f4e2d7R240-R246

What do you think of ignoring the events that have `getWheelDuration()=0`?

Reply:
Ah, I understand. I can vouch that the current code works on a magic trackpad, I just tested it locally.

Reply:
I thought the current code would fire unnecessary `getHorizontalScrollBar().setValue((int) (pos * chart.getWidth()));` events even for `=0` case

Reply:
Fair enough, I added a check for this